lib/faraday/adapter/net_http_persistent.rb

```diff
@@ -17,8 +17,11 @@ def with_net_http_connection(env)
             define_method(:password) { proxy[:password] }
           end if proxy[:user]
         end
-
-        yield Net::HTTP::Persistent.new 'Faraday', proxy_uri
+        if Net::HTTP::Persistent::VERSION >= '3'
```

Comment:
Our problem here is that they both (3.x and 2.x) return `-1` for that arity - they both default on each of the given arguments.

Inspired by this comment, I changed the test invocation to avoid testing for the version.